### PR TITLE
issue #328 (Nightly Draft)

### PR DESCRIPTION
**Pull request by Lomet**

### CODE REVIEW

File: contracts/AdvancedProviders/DelayVaultProvider/DelayVaultProvider.sol [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+import "./DelayVaultState.sol";
+
+contract DelayVaultProvider is DelayVaultState {
+    constructor(address _token, ILockDealNFT _nftContract, ProviderData[] memory _providersData) {
+        require(address(_token) != address(0x0), "invalid address");
+        require(address(_nftContract) != address(0x0), "invalid address");
+        require(_providersData.length <= 255, "too many providers");
+        name = "DelayVaultProvider";
+        Token = _token;
```
Comment (ashwinarora) [inline contracts/AdvancedProviders/DelayVaultProvider/DelayVaultProvider.sol:12]: DelayVaultV1 had support for all tokens. Do we want to restrict just 1 token?

File: contracts/SimpleProviders/Provider/ProviderState.sol [hunk 2]
```diff
@@ -2,14 +2,15 @@
...
 
 abstract contract ProviderState is IProvider {
     ///@dev Each provider sets its own name
-    string public name;
+    string public override name;
     ILockDealNFT public lockDealNFT;
 
     ///@dev each provider decides how many parameters it needs by overriding this function
-    function currentParamsTargetLenght() public view virtual returns (uint256) {
+    function currentParamsTargetLenght() public view virtual override returns (uint256) {
```
Comment (ashwinarora) [inline contracts/SimpleProviders/Provider/ProviderState.sol:13]: I think `override` does nothing here.

File: contracts/AdvancedProviders/DelayVaultProvider/HoldersSum.sol [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import "./MigratorV1/IMigrator.sol";
+
+abstract contract HoldersSum is ProviderModifiers, IDelayVaultData {
+    //this is only the delta
+    //the amount is the amount of the pool
+    // params[0] = startTimeDelta (empty for DealProvider)
+    // params[1] = endTimeDelta (only for TimedLockDealProvider)
+    event VaultValueChanged(address indexed Token, address indexed Owner, uint256 Amount);
+    mapping(address => uint256) public UserToAmount; //Each user got total amount
+    mapping(address => uint8) public UserToType; //Each user got type, can go up. wjem withdraw to 0, its reset
```
Comment (ashwinarora) [inline contracts/AdvancedProviders/DelayVaultProvider/HoldersSum.sol:15]: Ownership should be handled by `LockedDealNFT`, Not sure if DelayVault needs to know the owner.

**Review by ashwinarora — COMMENTED** [review]
Looks Good!

**Comment by Lomet** [comment]
moved to https://github.com/The-Poolz/DelayVaultProvider/pull/1

**Pull request closed without merging**